### PR TITLE
Enhancements to Storage and Filesystems documentation

### PR DESCRIPTION
**Pull request by tpdorsey**
Updates Storage and Filesystems documentation

- Adds notes about installing libraries used by some filesystems
- Adds missing GitHub and SMB docs
- Fixes docstring formatting
- Minor fixes and reorganization

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a `fix`, `feature`, `enhancement`, `docs`, `collections`, or `maintenance` label categorizing the change.

### CODE REVIEW

File: docs/concepts/filesystems.md [hunk 1]
```diff
@@ -143,52 +174,55 @@ block.save("dev")
...
+<div class="terminal">
 ```bash
-prefect deployment build path/to/flow.py:flow_name --name deployment_name --tag dev -sb az/dev
+prefect deployment build path/to/flow.py:flow_name --name deployment_name --tag dev -sb s3/dev
 ```
+</div>
 
-You need to install `adlfs` to use it.
-
+You need to install `s3fs`to use it.
```
Comment (billpalombi) [inline docs/concepts/filesystems.md:216]:
```suggestion
You need to install `s3fs` to use it.
```

**Review by billpalombi — APPROVED** [review]
LGTM

File: docs/concepts/deployments.md [hunk 2]
```diff
@@ -193,18 +196,30 @@ You may specify additional options to further customize your deployment.
 
 ### Block identifiers
 
-You can provide storage (`-sb`) and infrastructure block (`-ib`) identifiers in your `deployment build` command. The required format of a block type consists of the `block-type` and `block-name` in the format `block-type/block-name`. Block name is the name that you provided when creating the block. The block type is the same name as the underlying file system or infrastructure block class, but split into separate words combined with hyphens. Here are some examples that illustrate the pattern:
+When specifying a storage block with the `-sb` or `--storage-block` options, you may specify the block by passing its slug. The storage block slug is formatted as `block-type/block-name`.
```
Comment (discdiver) [inline docs/concepts/deployments.md:199]:
```suggestion
When specifying a storage block with the `-sb` or `--storage-block` flag, you may specify the block by passing its slug. The storage block slug is formatted as `block-type/block-name`. 
```

File: docs/concepts/deployments.md [hunk 3]
```diff
@@ -193,18 +196,30 @@ You may specify additional options to further customize your deployment.
...
-| `KubernetesJob` | `kubernetes-job` |
-| `Process` | `process` |
+For example, `s3/example-block` is the slug for an S3 block named `example-block`.
+
+In addition, when passing the storage block slug, you may pass just the block slug or the block slug and a path.
+
+- `block-type/block-name` indicates just the block, including any path included in the block configuration.
+- `block-type/block-name/path` indicates a storage path in addition to any path included in the block configuration.
+
+When specifying an infrastructure block with the `-ib` or `--infra-block` options, you specify the block by passing its slug. The infrastructure block slug is formatted as `block-type/block-name`.
```
Comment (discdiver) [inline docs/concepts/deployments.md:208]:
```suggestion
When specifying an infrastructure block with the `-ib` or `--infra-block` flag, you specify the block by passing its slug. The infrastructure block slug is formatted as `block-type/block-name`. 
```

File: docs/concepts/deployments.md [hunk 4]
```diff
@@ -193,18 +196,30 @@ You may specify additional options to further customize your deployment.
...
+| --- | --- | --- |
+| Azure | `Azure` | `azure` |
+| Docker Container | `DockerContainer` | `docker-container` |
+| GitHub | `GitHub` | `github` |
+| GCS | `GCS` | `gcs` |
+| Kubernetes Job | `KubernetesJob` | `kubernetes-job` |
+| Process | `Process` | `process` |
+| Remote File System | `RemoteFileSystem` | `remote-file-system` |
+| S3 | `S3` | `s3` |
+| SMB | `SMB` | `smb` |
```
Comment (discdiver) [inline docs/concepts/deployments.md:220]: Helpful table!

File: docs/concepts/filesystems.md [hunk 5]
```diff
@@ -8,20 +8,21 @@ tags:
     - RemoteFileSystem
 ---
 
-# File systems
+# Filesystems
 
-A file system block is an object which allows you to read and write data from paths. Prefect provides multiple built-in file system types that cover a wide range of use cases. 
+A filesystem block is an object which allows you to read and write data from paths. Prefect provides multiple built-in file system types that cover a wide range of use cases.
```
Comment (discdiver) [inline docs/concepts/filesystems.md:13]:
```suggestion
A filesystem block is an object that allows you to read and write data from paths. Prefect provides multiple built-in file system types that cover a wide range of use cases. 
```

File: docs/concepts/filesystems.md [hunk 6]
```diff
@@ -89,45 +89,76 @@ minio_block = RemoteFileSystem(
 minio_block.save("minio")
 ```
 
+## Azure
 
-## S3
-
-The `S3` file system block enables interaction with Amazon S3. Under the hood, `S3` uses [`s3fs`](https://s3fs.readthedocs.io/en/latest/).
+The `Azure` file system block enables interaction with Azure Datalake and Azure Blob Storage. Under the hood, `Azure` uses [`adlfs`](https://github.com/fsspec/adlfs).
```
Comment (discdiver) [inline docs/concepts/filesystems.md:94]:
```suggestion
The `Azure` file system block enables interaction with Azure Datalake and Azure Blob Storage. Under the hood, the `Azure` block uses [`adlfs`](https://github.com/fsspec/adlfs).
```

File: docs/concepts/filesystems.md [hunk 7]
```diff
@@ -89,45 +89,76 @@ minio_block = RemoteFileSystem(
...
+prefect deployment build path/to/flow.py:flow_name --name deployment_name --tag dev -sb az/dev
 ```
+</div>
 
-You need to install `s3fs`to use it.
+You need to install `adlfs` to use it.
+
+## GitHub
+
+The `GitHub` filesystem block enables interaction with GitHub repositories.
```
Comment (discdiver) [inline docs/concepts/filesystems.md:127]:
```suggestion
The `GitHub` filesystem block enables interaction with GitHub repositories. This block is currently read-only and limited to public repositories.
```

Comment (discdiver) [reply docs/concepts/filesystems.md:127]: I know there's an open issue for private repos and a hack, so might want to exclude the public repo section I added.

File: docs/concepts/filesystems.md (hunk 1 above)
Comment (discdiver) [inline docs/concepts/filesystems.md:216]:
```suggestion
You need to install `s3fs`to use this block.
```

File: docs/concepts/filesystems.md [hunk 8]
```diff
@@ -218,11 +255,15 @@ You need to install `smbprotocol` to use it.
...
+
+A Prefect installation and doesn't include filesystem-specific package dependencies such as `s3fs`, `gcsfs` or `adlfs`. This includes Prefect base Docker images.
+
+You must ensure that filesystem-specific libraries are installed in an execution environment where they will be used by flow runs.
 
-The core package and Prefect base images don't include filesystem-specific package dependencies such as `s3fs`, `gcsfs` or `adlfs`. To solve that problem in dockerized deployments, you can leverage the `EXTRA_PIP_PACKAGES` environment variable. Those dependencies will be installed at runtime within your Docker container or Kubernetes Job before the flow starts running. 
+In Dockerized deployments, you can leverage the `EXTRA_PIP_PACKAGES` environment variable. Those dependencies will be installed at runtime within your Docker container or Kubernetes Job before the flow starts running. 
 
-Here is an example showing how you can specify that in your deployment YAML manifest:
+Here is an example showing how you can specify that in your deployment YAML manifest, in this case `s3fs`:
```
Comment (discdiver) [inline docs/concepts/filesystems.md:266]:
```suggestion
Here is an example from a deployment YAML file showing how to specify the installation of `s3fs` from into your image:
```

File: docs/concepts/storage.md [hunk 9]
```diff
@@ -38,32 +38,59 @@ Anytime you call `prefect deployment build` without providing the `--storage-blo
...
+| --- | --- | --- |
+| [Local File System](/api-ref/prefect/filesystems/#prefect.filesystems.LocalFileSystem) | Store data in a run's local file system. | |
+| [Remote File System](/api-ref/prefect/filesystems/#prefect.filesystems.RemoteFileSystem) | Store data in a any filesystem supported by [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/). | |
+| [AWS S3 Storage](/api-ref/prefect/filesystems/#prefect.filesystems.S3) | Store data in an AWS S3 bucket. | [`s3fs`](https://s3fs.readthedocs.io/en/latest/) |
+| [Azure Storage](/api-ref/prefect/filesystems/#prefect.filesystems.Azure) | Store data in Azure Datalake and Azure Blob Storage. | [`adlfs`](https://github.com/fsspec/adlfs) |
+| [GitHub Storage](/api-ref/prefect/filesystems/#prefect.filesystems.GitHub) | Store data in a GitHub repository. | |
+| [Google Cloud Storage](/api-ref/prefect/filesystems/#prefect.filesystems.GCS) | Store data in a Google Cloud Platform (GCP) Cloud Storage bucket. | [`gcsfs`](https://gcsfs.readthedocs.io/en/latest/) |
+| [SMB](/api-ref/prefect/filesystems/#prefect.filesystems.SMB) | Store data in SMB shared network storage. | [`smbprotocol`](https://github.com/jborean93/smbprotocol) |
+
```
Comment (discdiver) [inline docs/concepts/storage.md:54]: Nice list! 🙂

**Review by discdiver — APPROVED** [review]
Looks good! Made minor edits you can ignore or keep as you like. Did not run code.

**Pull request merged**